src/libpriv/rpmostree-origin.cxx

```diff
@@ -115,23 +115,45 @@ rpmostree_origin_parse_keyfile (GKeyFile         *origin,
 
   ret->cached_unconfigured_state = g_key_file_get_string (ret->kf, "origin", "unconfigured-state", NULL);
 
-  g_autofree char *refspec = g_key_file_get_string (ret->kf, "origin", "refspec", NULL);
-  if (!refspec)
+  /* Note that the refspec type can be inferred from the key in the orgin file, where
+   * the `RPMOSTREE_REFSPEC_OSTREE_ORIGIN_KEY` and `RPMOSTREE_REFSPEC_OSTREE_BASE_ORIGIN_KEY` keys
+   * may correspond to either TYPE_OSTREE or TYPE_CHECKSUM (further classification is required), and 
+   * `RPMOSTREE_REFSPEC_CONTAINER_ORIGIN_KEY` always only corresponds to TYPE_CONTAINER. */
```

Comment:
These bits still feel a bit iffy to me; not sure if I interpreted https://github.com/coreos/rpm-ostree/pull/2940#issuecomment-870897552 correctly.